sdk/appconfiguration/azure-appconfiguration/azure/appconfiguration/_azure_appconfiguration_client.py

```diff
@@ -367,7 +369,7 @@ def set_configuration_setting(
                 key=key_value.key,  # type: ignore
                 label=key_value.label,
                 if_match=prep_if_match(configuration_setting.etag, match_condition),
-                if_none_match=prep_if_none_match(configuration_setting.etag, match_condition),
+                if_none_match=prep_if_none_match(etag or configuration_setting.etag, match_condition),
                 headers=custom_headers,
                 error_map=error_map,
             )
```

Comment:
Can't comment on the line, but while we're here we should make the same change for delete_configuration_setting. Once the pylint checker is fixed, that method will start throwing an error as well.

Reply:
Updated the same error in `delete_configuration_setting` and in other operations in this file(sync and async).